util/zenfs.cc

```diff
@@ -274,56 +274,52 @@ IOStatus zenfs_tool_copy_file(FileSystem *f_fs, std::string f, FileSystem *t_fs,
   std::unique_ptr<FSWritableFile> t_file;
   size_t buffer_sz = 1024 * 1024;
   uint64_t to_copy;
-  char *buffer;
- 
+  std::vector<char> buffer;
+
   fprintf(stdout, "%s\n", f.c_str());
- 
+
   s = f_fs->GetFileSize(f, iopts, &to_copy, &dbg);
   if (!s.ok()) { return s; }
 
   s = f_fs->NewSequentialFile(f, fopts, &f_file, &dbg);
   if (!s.ok()) { return s; }
-  
+
   s = t_fs->NewWritableFile(t, fopts, &t_file, &dbg);
   if (!s.ok()) { return s; }
 
   t_file->SetWriteLifeTimeHint(GetWriteLifeTimeHint(t));
 
-  buffer = (char *)malloc(buffer_sz);
-  if (buffer == nullptr) {
-    return IOStatus::IOError("Failed to allocate copy buffer");
-  }
+  buffer.resize(buffer_sz);
```

Comment:
It's nice to reduce the number of lines, but an allocation failure can still happen, right?